app/containers/Profile/ProfileForm.js

```diff
@@ -0,0 +1,128 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { Field, reduxForm } from 'redux-form/immutable';
+import { FormattedMessage } from 'react-intl';
+
+import messages from './messages';
+
+import {
+  AVATAR_FIELD,
+  DISPLAY_NAME_FIELD,
+  POSITION_FIELD,
+  COMPANY_FIELD,
+  ABOUT_FIELD,
+  LOCATION_FIELD,
+} from './constants';
+
+import renderTextInput from './renderTextInput';
+import renderTextarea from './renderTextarea';
+import renderFileInput from './renderFileInput';
+import renderLocationField from './renderLocationField';
+
+import { imageValidation, strLength20, strLength96 } from './validate';
+
+/* eslint-disable-next-line */
+let ProfileForm = props => {
+  const { handleSubmit, submitting, invalid, sendProps } = props;
+  const { ipfs } = sendProps.profile;
+
+  return (
+    <form onSubmit={handleSubmit(sendProps.saveProfile)}>
+      <div>
+        <Field
+          name={AVATAR_FIELD}
+          label={sendProps.translations[messages.avatarLabel.id]}
+          component={renderFileInput}
+          sendProps={sendProps}
+          validate={imageValidation}
+          warn={imageValidation}
+        />
+        <Field
+          name={DISPLAY_NAME_FIELD}
+          sendProps={sendProps}
+          component={renderTextInput}
+          label={sendProps.translations[messages.displayNameLabel.id]}
+          validate={strLength20}
+          warn={strLength20}
+        />
+        <Field
+          name={POSITION_FIELD}
+          sendProps={sendProps}
+          component={renderTextInput}
+          label={sendProps.translations[messages.positionLabel.id]}
+          validate={strLength20}
+          warn={strLength20}
+        />
+        <Field
+          name={COMPANY_FIELD}
+          sendProps={sendProps}
+          component={renderTextInput}
+          label={sendProps.translations[messages.companyLabel.id]}
+          validate={strLength20}
+          warn={strLength20}
+        />
+        <Field
+          name={ABOUT_FIELD}
+          component={renderTextarea}
+          sendProps={sendProps}
+          label={sendProps.translations[messages.aboutLabel.id]}
+          validate={strLength96}
+          warn={strLength96}
+        />
+        <Field
+          name={LOCATION_FIELD}
+          sendProps={sendProps}
+          component={renderLocationField}
+          label={sendProps.translations[messages.locationLabel.id]}
+        />
+      </div>
+      {sendProps.isOwner && (
+        <div>
+          <button
+            className="btn btn-success form-control"
+            disabled={
+              invalid ||
+              submitting ||
+              sendProps.loadingSaveProfile ||
```

Comment:
loadingSaveProfile is confusing name. Should it be just savingProfile?

Reply:
can i choose form with prefix loading? so i understand what this param does, for example isProfileLoading ?